ign_rviz_plugins/include/ignition/rviz/plugins/message_display_base.hpp

```diff
@@ -69,24 +82,18 @@ class MessageDisplay : public MessageDisplayBase
 
   virtual ~MessageDisplay() {}
 
-  // Documentation Inherited
-  virtual void initialize(rclcpp::Node::SharedPtr) {}
-
   /**
    * @brief ROS subscriber callback function
-   * @param[in] msg: ROS message type shared pointer
+   * @param[in] _msg: ROS message type shared pointer
    * @throws anything rclcpp::exceptions::throw_from_rcl_error can throw.
    */
-  virtual void callback(typename MessageType::SharedPtr) {}
+  virtual void callback(const typename MessageType::SharedPtr) {}
 
   /**
    * @brief Set ROS subscriber topic
    * @param[in] topic_name: ROS topic name
    */
-  virtual void setTopic(std::string) {}
-
-  // Documentation Inherited
-  virtual void setFrameManager(std::shared_ptr<common::FrameManager>) {}
+  virtual void setTopic(const std::string &) {}
```

Comment:
```suggestion
  virtual void setTopic(const std::string & topic_name) {}
```

Reply:
This is in the base class. Adding the param `toipc_name` here with a default implementation in the base class will lead to a warning saying that the param `topic_name` is not used. Same for the above.